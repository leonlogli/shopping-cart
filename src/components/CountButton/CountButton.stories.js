import React from 'react'

import CountButton from '.'

export default {
  title: 'CountButton',
}

export const Default = () => <CountButton>2</CountButton>
