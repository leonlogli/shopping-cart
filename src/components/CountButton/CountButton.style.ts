import styled from 'styled-components'

import { Box } from '..'

const CountButtonRoot = styled(Box)`
  background: #00000066;
  color: white;
  border-radius: 15px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  button {
    padding: 6px;
    margin: 0 8px;
  }

  svg {
    font-size: 16px;
  }
`

export { CountButtonRoot }
