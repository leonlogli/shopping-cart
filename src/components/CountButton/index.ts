export * from './CountButton'
export { default } from './CountButton'
