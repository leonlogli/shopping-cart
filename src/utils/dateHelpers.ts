const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
