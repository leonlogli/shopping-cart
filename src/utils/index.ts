export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
