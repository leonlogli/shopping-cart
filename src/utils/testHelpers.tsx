import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
