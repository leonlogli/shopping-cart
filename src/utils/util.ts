const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
