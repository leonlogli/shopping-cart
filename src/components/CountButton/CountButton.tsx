import React from 'react'
import AddIcon from '@material-ui/icons/Add'
import RemoveIcon from '@material-ui/icons/Remove'
import IconButton from '@material-ui/core/IconButton'

import { Text } from '../Typography'

import { CountButtonRoot } from './CountButton.style'
import { BoxProps } from '../Layout'

export type CountButtonProps = BoxProps & {
  onIncrement: () => void
  onDecrement: () => void
}

const CountButton = (props: CountButtonProps) => {
  const { onIncrement, onDecrement, children, ...other } = props

  return (
    <CountButtonRoot {...other}>
      <IconButton
        size="small"
        color="inherit"
        aria-label="remove"
        onClick={onDecrement}
      >
        <RemoveIcon />
      </IconButton>
      <Text variant="button">{children}</Text>
      <IconButton
        size="small"
        color="inherit"
        aria-label="add"
        onClick={onIncrement}
      >
        <AddIcon />
      </IconButton>
    </CountButtonRoot>
  )
}

export { CountButton }
export default CountButton
