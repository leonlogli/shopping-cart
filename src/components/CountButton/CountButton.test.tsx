import React from 'react'

import { mountWithTheme } from '../../utils'

import CountButton from './CountButton'

describe('<CountButton />', () => {
  const wrapper = mountWithTheme(
    <CountButton onIncrement={jest.fn()} onDecrement={jest.fn()}>
      2
    </CountButton>
  )

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
