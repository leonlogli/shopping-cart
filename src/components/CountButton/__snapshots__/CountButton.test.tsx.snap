// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountButton /> should match a snapshot 1`] = `
<CountButton
  onDecrement={[MockFunction]}
  onIncrement={[MockFunction]}
>
  <Styled(Component)>
    <ForwardRef
      className="eNdXh"
    >
      <Styled(Styled(MuiBox))
        className="eNdXh"
      >
        <Styled(MuiBox)
          className="dUNeVP eNdXh"
        >
          <div
            className="MuiBox-root MuiBox-root-1 dUNeVP eNdXh"
          >
            <WithStyles(ForwardRef(IconButton))
              aria-label="remove"
              color="inherit"
              onClick={[MockFunction]}
              size="small"
            >
              <ForwardRef(IconButton)
                aria-label="remove"
                classes={
                  Object {
                    "colorInherit": "MuiIconButton-colorInherit",
                    "colorPrimary": "MuiIconButton-colorPrimary",
                    "colorSecondary": "MuiIconButton-colorSecondary",
                    "disabled": "Mui-disabled",
                    "edgeEnd": "MuiIconButton-edgeEnd",
                    "edgeStart": "MuiIconButton-edgeStart",
                    "label": "MuiIconButton-label",
                    "root": "MuiIconButton-root",
                    "sizeSmall": "MuiIconButton-sizeSmall",
                  }
                }
                color="inherit"
                onClick={[MockFunction]}
                size="small"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  aria-label="remove"
                  centerRipple={true}
                  className="MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                  disabled={false}
                  focusRipple={true}
                  onClick={[MockFunction]}
                >
                  <ForwardRef(ButtonBase)
                    aria-label="remove"
                    centerRipple={true}
                    className="MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    disabled={false}
                    focusRipple={true}
                    onClick={[MockFunction]}
                  >
                    <button
                      aria-label="remove"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[MockFunction]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <ForwardRef>
                          <WithStyles(ForwardRef(SvgIcon))>
                            <ForwardRef(SvgIcon)
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 13H5v-2h14v2z"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef>
                      </span>
                      <WithStyles(memo)
                        center={true}
                      >
                        <ForwardRef(TouchRipple)
                          center={true}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(IconButton)>
            </WithStyles(ForwardRef(IconButton))>
            <Text
              variant="button"
            >
              <Styled(WithStyles(ForwardRef(Typography)))
                variant="button"
              >
                <WithStyles(ForwardRef(Typography))
                  className=""
                  variant="button"
                >
                  <ForwardRef(Typography)
                    className=""
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    variant="button"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-button"
                    >
                      2
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </Styled(WithStyles(ForwardRef(Typography)))>
            </Text>
            <WithStyles(ForwardRef(IconButton))
              aria-label="add"
              color="inherit"
              onClick={[MockFunction]}
              size="small"
            >
              <ForwardRef(IconButton)
                aria-label="add"
                classes={
                  Object {
                    "colorInherit": "MuiIconButton-colorInherit",
                    "colorPrimary": "MuiIconButton-colorPrimary",
                    "colorSecondary": "MuiIconButton-colorSecondary",
                    "disabled": "Mui-disabled",
                    "edgeEnd": "MuiIconButton-edgeEnd",
                    "edgeStart": "MuiIconButton-edgeStart",
                    "label": "MuiIconButton-label",
                    "root": "MuiIconButton-root",
                    "sizeSmall": "MuiIconButton-sizeSmall",
                  }
                }
                color="inherit"
                onClick={[MockFunction]}
                size="small"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  aria-label="add"
                  centerRipple={true}
                  className="MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                  disabled={false}
                  focusRipple={true}
                  onClick={[MockFunction]}
                >
                  <ForwardRef(ButtonBase)
                    aria-label="add"
                    centerRipple={true}
                    className="MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    disabled={false}
                    focusRipple={true}
                    onClick={[MockFunction]}
                  >
                    <button
                      aria-label="add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[MockFunction]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <ForwardRef>
                          <WithStyles(ForwardRef(SvgIcon))>
                            <ForwardRef(SvgIcon)
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef>
                      </span>
                      <WithStyles(memo)
                        center={true}
                      >
                        <ForwardRef(TouchRipple)
                          center={true}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(IconButton)>
            </WithStyles(ForwardRef(IconButton))>
          </div>
        </Styled(MuiBox)>
      </Styled(Styled(MuiBox))>
    </ForwardRef>
  </Styled(Component)>
</CountButton>
`;
